contrib/babelfishpg_tds/src/backend/tds/tdslogin.c

```diff
@@ -2259,6 +2259,29 @@ TdsSendLoginAck(Port *port)
 
 		TdsUTF8toUTF16StringInfo(&buf, default_server_name, prognameLen);
 		TdsPutbytes(buf.data, buf.len);
+		Assert(product_version != NULL);
+		if(pg_strcasecmp(product_version,"default") == 0)
+			version_pnt = ProcessVersionNumber(BABEL_COMPATIBILITY_VERSION);
+		else
+			version_pnt = ProcessVersionNumber(product_version);
+
+		MajorVersion = *(version_pnt + 0);
+		MinorVersion = *(version_pnt + 1);
+		MicroVersion = *(version_pnt + 2);
+		srvVersionBytes[0] = MajorVersion & 0xFF;
+		srvVersionBytes[1] = MinorVersion & 0xFF;
+
+		if (MicroVersion <= 0xFF)
+		{
+
+			srvVersionBytes[2] = 0x00;
+			srvVersionBytes[3] = MicroVersion & 0xFF;
+		}
+		else
+		{	
+			srvVersionBytes[2] = (MicroVersion >> 8) & 0xFF;
+			srvVersionBytes[3] = MicroVersion & 0xFF;
+		}
```

Comment:
1. We already have a similar code in `SetPreLoginResponseVal()`. To prevent this kind of issues in the future, let's make it as a common function.
2. As Alex suggested, could you considering adding a test for this?